### ISSUE TITLE
Via Stitching Generator ignores keepout areas

### ISSUE THREAD

**Issue by xobs** [jsreynaud/kicad-action-scripts#80]
The via stitching generator appears to ignore keepout areas.

For example, here is a U.Fl connector with a keepout area underneath it:

![image](https://github.com/user-attachments/assets/19b6a46d-a281-421c-9499-707f7d1fb0a5)

This array was generated with the defaults, except "Ignore areas on other layers" was selected because otherwise it generates 0 vias.

This is running kicad-action-scripts version 8.0.5 on kicad 8.0.4.

**Comment by 0x3333** [comment]
In my case, the issue is with Margin layer, which I use to keep cooper areas off.

**Comment by xobs** [comment]
Annoyingly, there's no way to manually remove the offending vias, since deleting it removes everything.

**Comment by jsreynaud** [comment]
> Annoyingly, there's no way to manually remove the offending vias, since deleting it removes everything.

Yes you can. All vias are inside a group. So ungroup them or enter in group then you will be able to delete individuals vias.

**Comment by xobs** [comment]
Oh, thank you! I didn't know that there was a workaround. I thought it generated a special kind of multi-via object that couldn't be manipulated, which is why the "Delete Vias" button just shows an alert.

Of course, once you ungroup things there's no way to undo the via stitching. Even so, that's a much better solution than to keep running the generator repeatedly while fiddling with the grid spacing in the hopes that I find a value that doesn't generate violations.
